### PR TITLE
Make entity context recognize schema defaults of associations.

### PR DESCRIPTION
**Pull request by ndm2**
The entity passed to `_schemaDefault()` is not a valid value to use
with `_getTable()`, hence the root table has been used for all schema
lookups.

### CODE REVIEW

File: src/View/Form/EntityContext.php [hunk 1]
```diff
@@ -275,16 +275,16 @@ public function val($field, $options = [])
     /**
      * Get default value from table schema for given entity field.
      *
-     * @param string $field Field name.
-     * @param \Cake\Datasource\EntityInterface $entity The entity.
+     * @param array $parts Each one of the parts in a path for a field name
      * @return mixed
      */
-    protected function _schemaDefault($field, $entity)
+    protected function _schemaDefault($parts)
```
[the remark this answers is not in the document]
Comment (ndm2) [reply src/View/Form/EntityContext.php]: Should it? Not sure what exactly the rules for protected APIs are... I can rebase on 3.next, or add backwards compatibility for the "broken" signature usage.

[the remark this answers is not in the document]
Comment (dereuromark) [reply src/View/Form/EntityContext.php]: It is a rather internal class. Overwriting it on project or plugin level is probably not happening a lot.

[the remark this answers is not in the document]
Comment (ADmad) [reply src/View/Form/EntityContext.php]: Change in signature of protected methods needs to go into next minor.

**Comment by ndm2** [comment]
aaaand i broke it...

**Comment by dereuromark** [comment]
Don't wry, those extra commits go away once master is merged into 3.next branch.

**Pull request merged**